bionic/persistence.py

```diff
@@ -57,24 +57,24 @@
 
 class PersistentCache:
     """
-    Provides a persistent mapping between Queries (things we could compute) and
-    saved Results (computed Queries).  You use it by getting a CacheAccessor
-    for your specific query, and then performing load/save operations on the
+    Provides a persistent mapping between Provenances (things we could compute) and
+    saved Results (computed provenances).  You use it by getting a CacheAccessor
+    for your specific Provenance, and then performing load/save operations on the
     accessor.
 
-    When looking up a Query, the cache searches for a saved artifact with a
-    matching Query.  The Query may not match exactly: each Query contains a
-    Provenance, which represents all the code and data used to compute a value,
-    and two Provenances can match at different levels of precision, from a
-    "functional" match to an "exact" one.  A functional match is sufficient to
-    treat two artifacts as interchangeable; the finer levels of matching are
-    only used by the "assisted versioning" system, which tries to detect
-    situations where a function's bytecode has changed but its version hasn't.
+    When looking up a Provenance, the cache searches for a saved artifact with a
+    matching Provenance. The Provenance may not match exactly: each which represents
```

Comment:
It seems like something is missing in this sentence. I think you need to replace "each which" with "each Provenance"?

Reply:
Thanks, good catch. I've changed the sentence structure.